packages/perseus/src/widgets/interactive-graphs/mafs-graph.tsx

```diff
@@ -70,79 +72,106 @@ export const MafsGraph = (props: MafsGraphProps) => {
-                            containerSizeClass={props.containerSizeClass}
-                            markings={props.markings}
-                        />
-                        {/* Locked layer */}
-                        {props.lockedFigures && (
-                            <GraphLockedLayer
-                                lockedFigures={props.lockedFigures}
+                        {props.markings === "graph" && (
+                            <>
+                                <AxisLabels />
+                                <AxisTickLabels />
+                            </>
+                        )}
+                        <Mafs
+                            preserveAspectRatio={false}
+                            viewBox={{
+                                x: state.range[X],
+                                y: state.range[Y],
+                                padding: 0,
+                            }}
+                            pan={false}
+                            zoom={false}
+                            width={width}
+                            height={height}
+                        >
+                            {/* Svg definitions to render only once */}
+                            <SvgDefs />
+                            {/* Background layer */}
+                            <Grid
+                                tickStep={props.step}
+                                gridStep={props.gridStep}
                                 range={state.range}
+                                containerSizeClass={props.containerSizeClass}
+                                markings={props.markings}
                             />
-                        )}
-                        {/* Protractor */}
-                        {props.showProtractor && <Protractor />}
-                        {/* Interactive layer */}
-                        {renderGraph({
-                            state,
-                            dispatch,
-                        })}
-                    </Mafs>
+                            {/* Locked layer */}
+                            {props.lockedFigures && (
+                                <GraphLockedLayer
+                                    lockedFigures={props.lockedFigures}
+                                    range={state.range}
+                                />
+                            )}
+                            {/* Protractor */}
+                            {props.showProtractor && <Protractor />}
+                            {/* Interactive layer */}
+                            {renderGraph({
+                                state,
+                                dispatch,
+                            })}
+                        </Mafs>
+                    </View>
                 </View>
+                {props.state.type === "point" &&
+                    props.state.numPoints === "unlimited" &&
+                    renderPointGraphControls({state, dispatch})}
```

Comment:
This feels a bit like points-specific logic is leaking out into this component. Currently, `renderGraph` is the thing that knows what components to render for each graph type. Maybe `renderGraph` should return two things: the graph itself, and the controls.